### PR TITLE
Fix that the SA1130 code fix crashes on an invalid event symbol

### PR DESCRIPTION
**Pull request by pdelvo**
This was found using the Tester on the monodevelop source.

### CODE REVIEW

File: StyleCop.Analyzers/StyleCop.Analyzers.CodeFixes/ReadabilityRules/SA1130CodeFixProvider.cs [hunk 1]
```diff
@@ -38,19 +38,34 @@ public override FixAllProvider GetFixAllProvider()
         }
 
         /// <inheritdoc/>
-        public override Task RegisterCodeFixesAsync(CodeFixContext context)
+        public override async Task RegisterCodeFixesAsync(CodeFixContext context)
         {
             foreach (var diagnostic in context.Diagnostics)
             {
+                if (await CanFixAsync(context, diagnostic).ConfigureAwait(false))
```
Comment (sharwell) [inline StyleCop.Analyzers/StyleCop.Analyzers.CodeFixes/ReadabilityRules/SA1130CodeFixProvider.cs]: ❗️ This is either a bad method name or the expression is inverted

```suggestion
                if (!(await CanFixAsync(context, diagnostic).ConfigureAwait(false)))
```

File: StyleCop.Analyzers/StyleCop.Analyzers.CodeFixes/ReadabilityRules/SA1130CodeFixProvider.cs [hunk 2]
```diff
@@ -221,7 +249,14 @@ private static async Task<Document> GetTransformedDocumentAsync(Document documen
 
             var anonymousMethod = (AnonymousMethodExpressionSyntax)syntaxRoot.FindNode(diagnostic.Location.SourceSpan, getInnermostNodeForTie: true);
 
-            var newSyntaxRoot = syntaxRoot.ReplaceNode(anonymousMethod, ReplaceWithLambda(semanticModel, anonymousMethod));
+            var newNode = ReplaceWithLambda(semanticModel, anonymousMethod);
+
+            if (newNode == null)
+            {
+                return document;
```
Comment (sharwell) [inline StyleCop.Analyzers/StyleCop.Analyzers.CodeFixes/ReadabilityRules/SA1130CodeFixProvider.cs]: ❓ Is this reachable considering the condition added before registering the code fix?

File: StyleCop.Analyzers/StyleCop.Analyzers.CodeFixes/ReadabilityRules/SA1130CodeFixProvider.cs [hunk 3]
```diff
@@ -247,7 +282,17 @@ protected override async Task<SyntaxNode> FixAllInDocumentAsync(FixAllContext fi
...
                 }
 
-                return syntaxRoot.ReplaceNodes(nodes, (originalNode, rewrittenNode) => ReplaceWithLambda(semanticModel, rewrittenNode));
+                return syntaxRoot.ReplaceNodes(nodes, (originalNode, rewrittenNode) =>
+                {
+                    var newNode = ReplaceWithLambda(semanticModel, rewrittenNode);
+
+                    if (newNode == null)
+                    {
+                        return null;
```
Comment (sharwell) [inline StyleCop.Analyzers/StyleCop.Analyzers.CodeFixes/ReadabilityRules/SA1130CodeFixProvider.cs]: ❓ Shouldn't this case be returning `rewrittenNode`?

File: StyleCop.Analyzers/StyleCop.Analyzers.CodeFixes/ReadabilityRules/SA1130CodeFixProvider.cs (hunk 2 above)
Comment (pdelvo) [reply StyleCop.Analyzers/StyleCop.Analyzers.CodeFixes/ReadabilityRules/SA1130CodeFixProvider.cs]: I dont think so

**Comment by pdelvo** [comment]
@sharwell I implemented your comments

**Pull request merged**